### PR TITLE
Editable Reminder

### LINKED ISSUE #11: Editing A Note
When viewing an existing reminder, the user can click on an “Edit” button where they can modify the title, notes, or date of the reminder.

* When editing a note, there is a “Save” button that commits the changes to the server.
* When the user hits the save button, they are returned to the original individual note path (ie: `reminders/1`).  

Things to keep in mind:
* What should your route look like to edit a reminder?

### PR DESCRIPTION
**Pull request by adam-rice**
@martensonbj @dylanavery720 @Tman22 

## Purpose

The feature allows the user to select an Edit button when viewing a specific reminder. Upon clicking edit the user is able to update the reminder contents via a form. After submission of the changes the route is directed to '/reminders'.

fixes #11 ✏️ 🖇 ☎️ 

## Approach

We created a new edit route with the /reminders/reminder route. By creating a new edit-reminder component we were able to link an edit-reminder hbs to an action called editReminder. EditReminder updates the model with model.save().

### Learning

We had to discover how to send the model down to the edit-reminder-form by listing model in the ember tag {{action 'editReminder' model on='submit'}}.  We also had to research .save() so that the store would update the targeted model.

### Follow-up tasks

[issue 12](https://github.com/turingschool-projects/1610-remember-2/issues/12)

### Screenshots

[After](http://imgur.com/EXlBJcN)

### CODE REVIEW

**Comment by adam-rice** [comment]
@Tman22 c8e4784 Adds testing :rhinoceros:

File: tests/acceptance/reminder-list-test.js [hunk 1]
```diff
@@ -62,3 +62,13 @@ test('if there are no reminders there should be text "Please add your first remi
...
   });
 });
+
+test('if edit is selected a form is rendered where the user can update their reminder', function(assert){
+  server.createList('reminder', 1);
+
+  visit('/reminders/1/edit');
+
+  andThen(function() {
+    assert.equal(find('form').length, 1);
```
Comment (Tman22) [inline tests/acceptance/reminder-list-test.js]: Can we do more here? Assert the model.title starts off as the editable text. Then change the input field and click save. Assert the current(url) is and that the record is clean.

Comment (Tman22) [inline tests/acceptance/reminder-list-test.js:72]: Can we do more here? Assert the model.title starts off as the value in the input field. Afterwards change the input field and click save. Assert the current(url) is correct and the record is clean. 

So the acceptance tests in this file are pretty shallow. An acceptance test is the walk through of your UI. 
For example: If you expect the user to create a reminder on 'reminders/new'; an acceptance test asserts that the user can fill in the inputs and click submit.

**Comment by adam-rice** [comment]
@Tman22 @martensonbj @dylanavery720 new test added to ensure content of reminder is being updated 35c85be

**Comment by adam-rice** [comment]
@Tman22 @dylanavery720 test added for creating a reminder 🐅

**Comment by Tman22** [comment]
Nice work!

**Pull request merged**